ophyd/sim.py

```diff
@@ -127,7 +127,10 @@ def __init__(self, func=None, *,
             # Initialize readback with a None value
             self._readback = None
         if loop is None:
-            loop = asyncio.get_event_loop()
+            try:
+                loop = asyncio.get_event_loop()
```

Comment:
under what conditions does this fail?

Reply:
If you try to instantiate it inside of a QThread or regular python Thread.